### ISSUE TITLE
Proposal: Lambda expression set-only properties

### ISSUE THREAD

**Issue by AustinBryan** [dotnet/csharplang#1615]
I know we almost never use set-only properties, and, because they have to be side effect driven, some people say that they should never be used and a method should be used instead. Regardless, here is an example of when I'd find it useful: 

```csharp
public class Cube
{
    public int X { get; private set; }
    public int Y { get; private set; }
    public int Z { get; private set; }

    public int Volume => X * Y * Z;

    // Current
    public int Scale { set => X = Y = Z = value; }
    // Proposal
    public int Scale <= X = Y = Z = value;
    // Or
    public int Scale <- X = Y = Z = value;
}
```

Ideally it would be `<=` to mirror the get-only `=>`, but if that gets confusing with the existing boolean operator, maybe something similar like `<-` .

**Comment by stakx** [comment]
Hmm, like you are saying, `set`-only properties are already a rare occurrence, and one that is not encouraged, so I am not sure your proposal to expand the language with a construct targeting them will meet much support.

That being said, I take issue with your symbols of choice, `<=` and `<-`, both of which I find problematic because they suggest a meaning other than what you intend. This is quite subjective, but I thought I'd mention it anyway:

I don't speak for the language design team, and I don't know the full story behind the `=>` symbol, but I've always understood `lhs => rhs` to mean "`lhs` maps to `rhs`" in a mathematical sense. I would probably read it the same way if its direction were reversed (`<=`).

In addition, using `<-` might be confusing to two groups of people: those coming from certain functional languages (for example, F#) where that symbol is used for assignment; and those coming from C++ who know IOStream's left-directed `<<`. In both cases, people might easily assume that `LHS <- rhs` means "`rhs` is being put into `LHS`".

**Comment by AustinBryan** [comment]
@stakx I see where your coming from. I see `lhs <= rhs` as `rhs` goes to `lhs`, which is kind of what's going on. The body is what happens when someone tries to put something inside `lhs` . 

At the same time, it isn't as necessary still as `=>` was because that had to have a `return` inside it, whereas `set` is only saving a few more characters. 

```csharp
Foo { get { return foo = value; } }   // only setting foo here to make an equal comparision
Foo => foo = value;                   // nearly half the size

Foo { set { foo = value; } }
Foo => foo = value;    // not as big of an improvement
```

Also, IIRC,  `Foo => _foo;` was allowed before `foo { get => _foo; }`, right? That would mean it was more needed at the time. 

I came across an instance the other day when I could've used a set-only, but then I found a better way to do it. Regardless, it's a feature I wouldn't mind.
